typesafety/test_interfaces/test_specific/test_failable/test_diverse_failable.yml

```diff
@@ -0,0 +1,145 @@
+
+
+- case: diverse_failable_inheritance_missing
+  disable_cache: true
+  main: |
+    from typing import TypeVar
+    from typing_extensions import final
+    from returns.interfaces.specific.failable import DiverseFailable2
+    from returns.primitives.hkt import SupportsKind2
+
+    _FirstType = TypeVar('_FirstType')
+    _SecondType = TypeVar('_SecondType')
+
+    @final
+    class MyClass(
+        SupportsKind2['MyClass', _FirstType, _SecondType],
+        DiverseFailable2[_FirstType, _SecondType],
+    ):
+        ...
+  out: |
+    main:10: error: Final class main.MyClass has abstract attributes "alt", "apply", "bind", "from_failure", "from_iterable", "from_value", "map", "rescue", "swap"
+
+
+- case: diverse_failable_inheritance_wrong2
+  disable_cache: true
+  main: |
+    from typing import TypeVar
+    from returns.interfaces.specific.failable import DiverseFailable2
+    from returns.primitives.hkt import SupportsKind2
+
+    _FirstType = TypeVar('_FirstType')
+    _SecondType = TypeVar('_SecondType')
+
+    class MyClass(
+        SupportsKind2['MyClass', _FirstType, _SecondType],
+        DiverseFailable2[_FirstType, _SecondType],
+    ):
+        def swap(self) -> MyClass[_FirstType, _SecondType]:
+            ...
+
+        @classmethod
+        def from_failure(
+            cls,
+            inner_value: _FirstType
+        ) -> MyClass[_FirstType, _FirstType]:
+            ...
+
+    x: MyClass[str, int]
+  out: |
+    main:12: error: Return type "MyClass[_FirstType, _SecondType]" of "swap" incompatible with return type "KindN[MyClass[_FirstType, _SecondType], _SecondType, _FirstType, NoReturn]" in supertype "DiverseFailableN"
+    main:16: error: Argument 1 of "from_failure" is incompatible with supertype "DiverseFailableN"; supertype defines the argument type as "_SecondType"
+    main:16: error: Return type "MyClass[_FirstType, _FirstType]" of "from_failure" incompatible with return type "KindN[MyClass[_FirstType, _SecondType], _FirstType, _SecondType, NoReturn]" in supertype "DiverseFailableN"
+    main:16: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
+    main:16: note: This violates the Liskov substitution principle
+
+
+- case: diverse_failable_inheritance_wrong3
+  disable_cache: true
+  main: |
+    from typing import TypeVar
+    from returns.interfaces.specific.failable import DiverseFailable3
+    from returns.primitives.hkt import SupportsKind3
+
+    _FirstType = TypeVar('_FirstType')
+    _SecondType = TypeVar('_SecondType')
+    _ThirdType = TypeVar('_ThirdType')
+
+    class MyClass(
+        SupportsKind3['MyClass', _FirstType, _SecondType, _ThirdType],
+        DiverseFailable3[_FirstType, _SecondType, _ThirdType],
+    ):
+        def swap(self) -> MyClass[_FirstType, _SecondType, _ThirdType]:
+            ...
+
+        @classmethod
+        def from_failure(
+            cls,
+            inner_value: _FirstType
```

Comment:
Try `inner_value: _SecondType` here, it should be incorrect.

Reply:
Done!